src/ext/java/org/opentripplanner/ext/siri/SiriFuzzyTripMatcher.java

```diff
@@ -36,23 +36,37 @@
  * process will always be applied even in places where you have good quality IDs in SIRI data and
  * don't need it - we'd have to add a way to disable it.
  * <p>
- * Several instances of this SiriFuzzyTripMatcher may appear in different SIRI updaters, but they
- * all share a common set of static Map fields. We generally don't advocate using static fields in
- * this way, but as part of a sandbox contribution we are maintaining this implementation.
+ * The same instance of this SiriFuzzyTripMatcher may appear in different SIRI updaters. Be sure
+ * to fetch the instance at during the setup of the updaters, the initialization is not thread-safe.
  */
 public class SiriFuzzyTripMatcher {
 
   private static final Logger LOG = LoggerFactory.getLogger(SiriFuzzyTripMatcher.class);
-  private static final Map<String, Set<Trip>> mappedTripsCache = new HashMap<>();
-  private static final Map<String, Set<Trip>> mappedVehicleRefCache = new HashMap<>();
-  private static final Map<String, Set<Route>> mappedRoutesCache = new HashMap<>();
-  private static final Map<String, Set<Trip>> start_stop_tripCache = new HashMap<>();
-  private static final Map<String, Trip> vehicleJourneyTripCache = new HashMap<>();
-  private static final Set<String> nonExistingStops = new HashSet<>();
 
+  private static SiriFuzzyTripMatcher instance;
+
+  private final Map<String, Set<Trip>> mappedTripsCache = new HashMap<>();
+  private final Map<String, Set<Trip>> mappedVehicleRefCache = new HashMap<>();
+  private final Map<String, Set<Route>> mappedRoutesCache = new HashMap<>();
+  private final Map<String, Set<Trip>> start_stop_tripCache = new HashMap<>();
+  private final Map<String, Trip> vehicleJourneyTripCache = new HashMap<>();
+  private final Set<String> nonExistingStops = new HashSet<>();
   private final TransitService transitService;
 
-  public SiriFuzzyTripMatcher(TransitService transitService) {
+  /**
+   * Factory method used to crete only one instance.
+   * <p>
+   * THIS METHOD IS NOT THREAD-SAFE AND SHOULD BE CALLED DURING THE
+   * INITIALIZATION PROCESS.
+   */
+  public static SiriFuzzyTripMatcher of(TransitService transitService) {
```

Comment:
Why not just make this `synchronized`?

Reply:
It should be called during init, never during updates - so it should not be synchronized either.